### PR TITLE
IAM pagination support implemented

### PR DESCRIPTION
**Pull request by andrey-smirnov**
IAM requests with pagination were not supported in erlcloud, only first 100 elements were returned.
This PR fixes that issue by introducing Marker field in result of IAM list requests and separate *_all calls which retrieves all available elements in serie of sequential requests.

### CODE REVIEW

**Comment by kkuzmin** [comment]
@andrey-smirnov thank you for fixing specs.

**Comment by kkuzmin** [comment]
Please, add more tests with `meck:sequence()` responses to cover new list..._all functionality.

File: src/erlcloud_iam.erl [hunk 1]
```diff
@@ -74,15 +88,15 @@ configure(AccessKeyID, SecretAccessKey, Host) ->
...
 get_user() -> get_user([]).
--spec(get_user/1 :: (string() | aws_config()) -> proplist()).
+-spec(get_user/1 :: (string() | aws_config()) -> {ok, proplist()}).
 get_user(#aws_config{} = Config) ->
     get_user("", Config);
 get_user(UserName) ->
     get_user(UserName, default_config()).
 
--spec(get_user/2 :: (string(), aws_config()) -> proplist()).
+-spec(get_user/2 :: (string(), aws_config()) -> {ok, proplist()}).
```
Comment (motobob) [inline src/erlcloud_iam.erl]: where are all `| {error, any()}` ?

File: src/erlcloud_iam.erl [hunk 2]
```diff
@@ -492,10 +716,27 @@ iam_query(Config, Action, Params, ApiVersion) ->
                                   "/", QParams, "iam", Config).
 
 iam_query(Config, Action, Params, ItemPath, DataTypeDef) ->
-    case iam_query(Config, Action, Params) of
-        {ok, Doc} ->
-            Items = xmerl_xpath:string(ItemPath, Doc),
-            {ok, [extract_values(DataTypeDef, Item) || Item <- Items]};
+    iam_query(Config, Action, Params, ItemPath, DataTypeDef, false).
+
+iam_query(Config, Action, Params, ItemPath, DataTypeDef, All) ->
```
Comment (motobob) [inline src/erlcloud_iam.erl]: i'm **not** ok with this implementation with the last `true|false` argument.

File: src/erlcloud_iam.erl [hunk 3]
```diff
@@ -555,6 +796,24 @@ extract_account_summary(Item) ->
               end
       end, [], Entries).
 
+
+make_response(Xml, Result) ->
```
Comment (motobob) [inline src/erlcloud_iam.erl]: this fun belongs to utils either and i see copy-paste there

File: src/erlcloud_iam.erl [hunk 4]
```diff
@@ -74,15 +88,15 @@ configure(AccessKeyID, SecretAccessKey, Host) ->
 %
 % Users API
 %
--spec(get_user/0 :: () -> proplist()).
+-spec(get_user/0 :: () -> {ok, proplist()}).
 get_user() -> get_user([]).
--spec(get_user/1 :: (string() | aws_config()) -> proplist()).
+-spec(get_user/1 :: (string() | aws_config()) -> {ok, proplist()}).
```
Comment (motobob) [inline src/erlcloud_iam.erl]: this interface change from `proplists()` to `{ok, proplists()}` will brake a lot of things!

Comment (andrey-smirnov) [inline src/erlcloud_iam.erl]: specs were not updated for ages, function actually returns {ok, proplist{}}.

Comment (motobob) [inline src/erlcloud_iam.erl]: understood

**Comment by motobob** [comment]
looks way better. @kkuzmin waiting for your feedback before merge

**Comment by kkuzmin** [comment]
+1. Thanks for tests.

**Pull request merged**